lib/chain.js

```diff
@@ -24,7 +24,7 @@ module.exports = class SetupChain {
     this.parser = new Parser()
     this.lexer = new Lexer()
     this.visitor = new Visitor()
-    this.state = state || {} // runtime execution results
+    this.state = state ? {...state} : {} // runtime execution results
```

Comment:
I saw your comment about this in the channel - what's the reasoning behind shallow cloning? As a library I think it'd be preferable to not do this, and allow the user to clone in whatever way makes sense for their use-case before they pass in initial `state`.

Reply:
my reasoning was within a test, the common case is to pass the result of one chain to the next. If I make a global state and re-use it. other chains may modify it in way I hadn't expected. but I guess you could just as well pass `new Chain({...state})`

Reply:
our most common actions are generally building objects though, so shallow cloning `state` wouldn't (usually) help avoid that scenario.

this is definitely an important point though, and maybe something we could document and/or provide an example of in the readme

Reply:
it would avoid clobbering top level keys in a shared state at the least. I'm not married to this, it just struck me as a footgun when re-using the state. Passing this to chain B will modify the state of Chain A. I wouldn't really expect that.

Reply:
I will admit I don't think I've ever passed state from one chain to the next, so I never considered this before 😄 my only other argument for leaving it up to the user is that if you _wanted_ to mutate top-level primitive types for the initial state (for some reason), you couldn't.